source-code/sdk-js/src/helpers/ast.ts

```diff
@@ -0,0 +1,108 @@
+import type { CallExpression, Node, Program } from "estree"
+import { walk } from "estree-walker"
+import { get } from "lodash"
+
+type wrapVariableDeclaration = (
+	sourceAst: Program,
+	searchIdentifier: string,
+	identifier: string,
+) => Promise<Program>
+
+type insertAst = (
+	sourceAst: Program,
+	ast: Node,
+	position:
+		| { after?: never; before: [string, string, ...string[]] }
+		| { after: [string, string, ...string[]]; before?: never },
+) => Promise<Program>
+
+const wrapWithCallExpression = (n: Node, identifier: string) => {
+	return {
+		type: "CallExpression",
+		callee: {
+			type: "Identifier",
+			name: identifier,
+		},
+		arguments: [structuredClone(n)],
+		optional: false,
+	} as CallExpression
+}
+
+/**
+ * Finds a variable declaration with a given name in sourceAst and wraps the corresponding declarator with the given identifier function call.
+ * Is immutable towards it's inputs, meaning it DOESN'T MANIPULATE THEM.
+ *
+ * wrapVariableDeclaration(ast, "search", "wrap")` with an ast for the code `const search = () => {}` will result in `const search = wrap(() => {})`
+ *
+ * @param sourceAst The ast to operate on
+ * @param searchIdentifier The name of the variable declaration whose declarator should be wrapped
+ * @param identifier The function identifier with which the declarator should be wrapped.
+ * @returns A promise that resolves to the resulting ast or rejects if the searched for declarator can't be found.
+ */
+export const wrapVariableDeclaration = (async (sourceAst, searchIdentifier, identifier) => {
+	const sourceAstClone = structuredClone(sourceAst)
+	let found = false
+	walk(sourceAstClone, {
+		enter(node) {
+			if (
+				node.type === "VariableDeclarator" &&
+				node.id.type === "Identifier" &&
+				node.id.name === searchIdentifier &&
+				node.init
+			) {
+				found = true
+				const newNode = structuredClone(node)
+				newNode.init = wrapWithCallExpression(node.init, identifier)
+				this.replace(newNode)
+				this.skip()
+			}
+		},
+	})
+	if (!found) throw new Error("Couldn't find variable declarator.")
```

Comment:
we should use `[data, exception]` tuples instead of throwing errors. Take a look at `source-code/core/src/utilities/result.ts`

Reply:
Will fix